### ISSUE TITLE
3.8 new JAR files in java-modules directory

### ISSUE THREAD

**Issue by czanik** [balabit/syslog-ng#1118]
After updating my openSUSE package to latest git head, some new JAR files appeared under /usr/lib/syslog-ng/java-modules What are these? Are all of these necessary? 

```
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/commons-codec-1.9.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/commons-lang3-3.4.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/commons-logging-1.2.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/gson-2.6.2.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/guava-19.0.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/httpasyncclient-4.1.1.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/httpclient-4.5.2.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/httpcore-4.4.4.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/httpcore-nio-4.4.4.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/jest-2.0.2.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/jest-common-2.0.2.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/slf4j-api-1.7.13.jar
```

(My guess is, that these are dependency for the Elasticsearch HTTP client and related to https://github.com/balabit/syslog-ng/pull/1053 )

**Comment by lbudai** [comment]
@czanik: as you know we are using gradle for dependency management, and if gradle says that these are the dependencis, then the answer is yes, all of them are needed.
Is it a bug or can we close this issue?

**Comment by czanik** [comment]
Until now only a few files were installed here by syslog-ng:
```
%dir %{_libdir}/syslog-ng/java-modules
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/syslog-ng-core.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/dummy.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/elastic.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/elastic-v2.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/hdfs.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/http.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/kafka.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/log4j-1.2.16.jar
%attr(755,root,root) %{_libdir}/syslog-ng/java-modules/syslog-ng-common.jar
```
The rest of the jar files had to be loaded separately from other directories. That's why I don't know if it's a bug, or a policy change.

**Comment by lbudai** [comment]
For previous java destination drivers we forced the user to install the dependencies (it was not a big deal as those libs are part of the product they use, like Elasticsearch, HDFS, Kafka). This time the situation is little bit different: these libraries are not easily available to users (they can write their own mvn/gradle script after they checked what version they need, or, we can document it, but the feature is then not an OOB feature).  Maybe we could move them to a separate directory under java-modules.

**Comment by lbudai** [comment]
@czanik : so, will you need any other info to make a decision whether it is a problem to you or can we close this?

**Comment by lbudai** [comment]
@czanik : then we should have to re-open every issue that will be mentioned in the release notes.
My suggestion: using the 3.8 label and at the end we will check these items.
( You could help us in labeling those issues that should be also mentioned in the release notes. )

**Comment by lbudai** [comment]
@czanik : FYI : PR #1155 
I moved the jest-client jars into a separate directory inside java-modules.

**Comment by MrAnno** [comment]
Closing the issue due to inactivity.
The problem has been resolved in #1155.
